Add CIS Benchmark documentation to KubeOne (#1637)
kubermatic/docs: content/kubeone/main/guides/cis-benchmarking/cis1.8-kubeone1.7-k8s1.27/_index.en.md

**Result:** 🟢 Pass

---
#### 1.1.2: Ensure that the API server pod specification file ownership is set to root:root (Automated)

**Result:** 🟢 Pass

---
#### 1.1.3: Ensure that the controller manager pod specification file permissions are set to 600 or more restrictive (Automated)

**Result:** 🟢 Pass

---
#### 1.1.4: Ensure that the controller manager pod specification file ownership is set to root:root (Automated)

**Result:** 🟢 Pass

---
#### 1.1.5: Ensure that the scheduler pod specification file permissions are set to 600 or more restrictive (Automated)

**Result:** 🟢 Pass

---
#### 1.1.6: Ensure that the scheduler pod specification file ownership is set to root:root (Automated)

**Result:** 🟢 Pass

---
#### 1.1.7: Ensure that the etcd pod specification file permissions are set to 600 or more restrictive (Automated)

**Result:** 🟢 Pass

---
#### 1.1.8: Ensure that the etcd pod specification file ownership is set to root:root (Automated)

**Result:** 🟢 Pass

---
#### 1.1.9: Ensure that the Container Network Interface file permissions are set to 600 or more restrictive (Manual)

**Result:** 🟡 Warn

> _The issue is under investigation to provide a fix in a future KubeOne release_

Reviewer: What's the difference between Warn and Fail? Should we maybe explain that the beginning of the document?

Author: Actually, this is explained on top of the document, but I can update the wording.

Reviewer: No, I just missed it for some reason. Sorry 🤦🏻

**Result:** 🟢 Pass

---
#### 1.1.14: Ensure that the admin.conf file ownership is set to root:root (Automated)

**Result:** 🟢 Pass

---
#### 1.1.15: Ensure that the scheduler.conf file permissions are set to 600 or more restrictive (Automated)

**Result:** 🟢 Pass

---
#### 1.1.16: Ensure that the scheduler.conf file ownership is set to root:root (Automated)

**Result:** 🟢 Pass

---
#### 1.1.17: Ensure that the controller-manager.conf file permissions are set to 600 or more restrictive (Automated)

**Result:** 🟢 Pass

---
#### 1.1.18: Ensure that the controller-manager.conf file ownership is set to root:root (Automated)

**Result:** 🟢 Pass

---
#### 1.1.19: Ensure that the Kubernetes PKI directory and file ownership is set to root:root (Automated)

**Result:** 🟢 Pass

---
#### 1.1.20: Ensure that the Kubernetes PKI certificate file permissions are set to 600 or more restrictive (Manual)

**Result:** 🟢 Pass

---
#### 1.1.21: Ensure that the Kubernetes PKI key file permissions are set to 600 (Manual)

**Result:** 🟢 Pass

---
### 1.2. API Server
#### 1.2.1: Ensure that the --anonymous-auth argument is set to false (Manual)

**Result:** 🔵 Not Applicable

> **Details:** This is mitigated by RBAC, please see https://kubernetes.io/docs/reference/access-authn-authz/authentication/#anonymous-requests

Reviewer: I'm not sure this is sufficient. At best, we'd disable anonymous auth completely. Do we have any idea why this isn't set by KubeOne/kubeadm?

---
#### 1.2.2: Ensure that the --token-auth-file parameter is not set (Automated)

**Result:** 🟢 Pass

---
#### 1.2.3: Ensure that the --DenyServiceExternalIPs is set (Manual)

**Result:** 🔵 Not Applicable

> **Details:** DenyServiceExternalIPs is not enabled by default, and it is not in the enabled plugins list

Reviewer: As a user, I don't understand why this is not applicable. Can we please elaborate what this is and why we don't think it's needed?

**Result:** 🔵 Not Applicable

**Details:** AlwaysPullImages admission control will force all images to be pulled every time, it's not efficient for all users. This can be performed by a Kyverno or OPA policy

---
#### 1.2.12: Ensure that the admission control plugin SecurityContextDeny is set if PodSecurityPolicy is not used (Manual)

**Result:** 🔵 Not Applicable

**Details:** SecurityContextDeny admission control plugin is deprecated as of Kubernetes 1.27: https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#securitycontextdeny

---
#### 1.2.13: Ensure that the admission control plugin ServiceAccount is set (Automated)

**Result:** 🟢 Pass

---
#### 1.2.14: Ensure that the admission control plugin NamespaceLifecycle is set (Automated)

**Result:** 🟢 Pass

---
#### 1.2.15: Ensure that the admission control plugin NodeRestriction is set (Automated)

**Result:** 🔴 Fail

_The issue is under investigation to provide a fix in a future KubeOne release_

---
#### 1.2.16: Ensure that the --profiling argument is set to false (Automated)

**Result:** 🟢 Pass

---
#### 1.2.17: Ensure that the --audit-log-path argument is set (Automated)

**Result:** 🟠 Configurable

> **Details:** Audit logging is not enabled by default, it can be configured as described here: https://docs.kubermatic.com/kubeone/v1.7/tutorials/creating-clusters-oidc/#audit-logging

Reviewer: The URLs should be `ref` instead of absolute URL, so they link to the right version of the documentation. Here's an example: https://github.com/kubermatic/docs/blob/master/content/kubeone/main/_index.en.md?plain=1#L78

**Result:** 🟢 Pass

---
### 4.2. Kubelet
#### 4.2.1: Ensure that the --anonymous-auth argument is set to false (Automated)

**Result:** 🟢 Pass

---
#### 4.2.2: Ensure that the --authorization-mode argument is not set to AlwaysAllow (Automated)

**Result:** 🟢 Pass

---
#### 4.2.3: Ensure that the --client-ca-file argument is set as appropriate (Automated)

**Result:** 🟢 Pass

---
#### 4.2.4: Verify that the --read-only-port argument is set to 0 (Manual)

**Result:** 🟢 Pass

---
#### 4.2.5: Ensure that the --streaming-connection-idle-timeout argument is not set to 0 (Manual)

**Result:** 🟢 Pass

---
#### 4.2.6: Ensure that the --make-iptables-util-chains argument is set to true (Automated)

**Result:** 🟢 Pass

---
#### 4.2.7: Ensure that the --hostname-override argument is not set (Manual)

**Result:** 🟢 Pass

---
#### 4.2.8: Ensure that the eventRecordQPS argument is set to a level which ensures appropriate event capture (Manual)

**Result:** 🟢 Pass

---
#### 4.2.9: Ensure that the --tls-cert-file and --tls-private-key-file arguments are set as appropriate (Manual)

**Result:** 🔵 Not Applicable

> **Details:** This is a manual check, `--tls-cert-file` and `--tls-private-key-file` options are provided to Kubelet

Reviewer: I think this should be Pass. We can still keep the details.

Author: Ok, `not applicable` because of my script generation. I will update change it to `pass`

+++

> title = "Benchmark on KubeOne-1.7.3 and Kubernetes-1.27"

Reviewer 2: We don't use hyphen to separate product name and version. Also, I'd consider rotating the Kubernetes version and the KubeOne version, and dropping the KubeOne patch release:
```suggestion
title = "Benchmark on Kubernetes 1.27 with KubeOne 1.7"
```

date = 2024-03-06T12:01:00+02:00
+++

> This benchmark guide helps you evaluate the security of a KubeOne cluster against each control in the CIS Kubernetes Benchmark.

Reviewer 2: Some minor nits:
```suggestion
This guide helps you evaluate the security of a Kubernetes cluster created using KubeOne against each control in the CIS Kubernetes Benchmark.
```

This guide corresponds to the following versions of KubeOne, CIS Benchmarks, and Kubernetes:

| KubeOne Version  | Kubernetes Version | CIS Benchmark Version |
| ---------------- | ------------------ | --------------------- |

> | 1.7.3               | 1.27                 | CIS-1.8                    |

Reviewer 2: Nit: please add the Kubernetes patch release that you used

## Testing Methodology

Each control in the CIS Kubernetes Benchmark was evaluated against a KubeOne cluster that was configured according to the accompanying hardening guide.

These are the possible results for each control:

> 🟢 **Pass:** The KubeOne cluster passes the audit outlined in the benchmark.

Reviewer 2: ```suggestion
🟢 **Pass:** The KubeOne cluster passes the audit/control outlined in the benchmark.
```

🔵 **Pass (Additional Configuration Required):** The KubeOne cluster passes the audit outlined in the benchmark with some extra configuration. The documentation is provided.

> 🔴 **Fail:** The control will be fixed in a future KubeOne release.

Reviewer 2: ```suggestion
🔴 **Fail:** The audit/control will be fixed in a future KubeOne release.
```

#### 1.2.1: Ensure that the --anonymous-auth argument is set to false (Manual)

**Result:** 🔴 Fail

_The issue is under investigation to provide a fix in a future KubeOne release_

---
#### 1.2.2: Ensure that the --token-auth-file parameter is not set (Automated)

**Result:** 🟢 Pass

---
#### 1.2.3: Ensure that the --DenyServiceExternalIPs is set (Manual)

**Result:** 🔴 Fail

_The issue is under investigation to provide a fix in a future KubeOne release_

---
#### 1.2.4: Ensure that the --kubelet-client-certificate and --kubelet-client-key arguments are set as appropriate (Automated)

**Result:** 🟢 Pass

---
#### 1.2.5: Ensure that the --kubelet-certificate-authority argument is set as appropriate (Automated)

**Result:** 🟢 Pass

---
#### 1.2.6: Ensure that the --authorization-mode argument is not set to AlwaysAllow (Automated)

**Result:** 🟢 Pass

---
#### 1.2.7: Ensure that the --authorization-mode argument includes Node (Automated)

**Result:** 🟢 Pass

---
#### 1.2.8: Ensure that the --authorization-mode argument includes RBAC (Automated)

**Result:** 🟢 Pass

---
#### 1.2.9: Ensure that the admission control plugin EventRateLimit is set (Manual)

**Result:** 🟢 Pass

> **Details:** EventRateLimit admission control plugin in in Alpha state, please see [here][eventratelimit]

Reviewer 2: I was thinking more about this audit and I think it's actually a fail. We don't enable this admission control plugin, CIS requires it and doesn't go into details if it's alpha or not. It's up to us to decide if we want to support an alpha feature or not (we have always been towards not supporting alpha features and we can check with @csengerszabo if it's safe to state so in this document)

---
#### 1.2.10: Ensure that the admission control plugin AlwaysAdmit is not set (Automated)

**Result:** 🟢 Pass

---
#### 1.2.11: Ensure that the admission control plugin AlwaysPullImages is set (Manual)

**Result:** 🔴 Fail

_The issue is under investigation to provide a fix in a future KubeOne release_

---
#### 1.2.12: Ensure that the admission control plugin SecurityContextDeny is set if PodSecurityPolicy is not used (Manual)

**Result:** 🟢 Pass

> **Details:** SecurityContextDeny admission control plugin is deprecated as of [Kubernetes 1.27][securitycontextdeny]

Reviewer 2: When I think better, this is also a fail, for the same reason as the previous comment. We don't enable this admission control plugin and that's all what CIS cares about. We can say that this audit failed but that we don't support enabling this admission control plugin because the official Kubernetes docs state that it shouldn't be used at all and that it's deprecated.
